commercial/app/model/commercial/events/LiveEventAgent.scala

```diff
@@ -64,10 +62,12 @@ object LiveEventAgent extends ExecutionContexts with Logging {
 
     val liveEvents =
       for{
-        eventImages <- fetchAndParseLiveEventImages
+        eventMembershipInfo <- fetchAndParseLiveEventsMembershipInfo
         ParsedFeed(events, _) <- Eventbrite.parsePagesOfEvents(feedMetaData, feedContent)
-      } yield events
-            .map ( event => LiveEvent(event, eventImages find (_.eventId == event.id)) )
+      } yield events flatMap { event =>
```

Comment:
I could do it if I repeat the guard condition, but not sure it is much of an improvement:

```
val liveEvents =
      for{
        eventMembershipInfo <- fetchAndParseLiveEventsMembershipInfo
        ParsedFeed(events, _) <- Eventbrite.parsePagesOfEvents(feedMetaData, feedContent)
        event <- events
        if eventMembershipInfo.exists(_.id == event.id)
      } yield eventMembershipInfo.find(_.id == event.id).map(LiveEvent(event, _))
```